Integrations/integration-PagerDuty.yml

```diff
@@ -0,0 +1,475 @@
+- display: Service Key (for triggering events only)
+  name: ServiceKey
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: Fetch incidents
+  name: isFetch
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Incident type
+  name: incidentType
+  defaultvalue: ""
+  type: 13
+  required: false
+script:
+  script: |
+    ''' IMPORTS '''
+    import os
+    import json
+    import requests
+    import datetime
+
+    ''' GLOBAL VARS '''
+    USE_SSL = True
+
+    API_KEY = demisto.params()['APIKey']
+    SERVICE_KEY = demisto.params()['ServiceKey']
+
+    SERVER_URL = "https://api.pagerduty.com/"
+
+    DEFAULT_HEADERS = {
+                          'Authorization': 'Token token=' + API_KEY,
+                          'Accept': 'application/vnd.pagerduty+json;version=2'
+                      }
+
+    '''SUFFIX ENDPOINTS'''
+    GET_SCHEDULES_SUFFIX = "schedules/"
+    CREATE_INCIDENT_SUFFIX = "incidents"
+    ON_CALL_BY_SCHEDULE_SUFFIX = "schedules/{0}/users"
+    ON_CALLS_USERS_SUFFIX = "oncalls?include%5B%5D=users"
+    GET_INCIDENTS_SUFFIX = "incidents?include%5B%5D=assignees"
+
+    '''TABLE NAMES'''
+    SCHEDULES = "All Schedules"
+    TRIGGER_EVENT = 'Trigger Event'
+    USERS_ON_CALL = 'Users On Call'
+    USERS_ON_CALL_NOW = 'Users On Call Now'
+
+    '''TABLE HEADERS'''
+    SCHEDULES_HEADERS = ["ID", "Name", "Today", "Time Zone", "Escalation Policy", "Escalation Policy ID"]
+    USERS_ON_CALL_NOW_HEADERS = ["ID", "Email", "Name", "Role", "Color", "User Url", "Avatar Url", "Time Zone"]
+    INCIDENTS_HEADERS = ["ID", "Incident Number", "Created On", "Status", "Html Url", "Service ID", "Service Name", "Service Html Url", "Assigned To User",
+                         "Escalation Policy", "Escalation Policy ID", "Trigger Details Html Url", "Trigger Type", "Last Status Change On", "Last Status Change By",
+                         "Number Of Escalations", "Resolved By User", "Assigned To", "Summary"]
+
+    ''' HELPER FUNCTIONS '''
+    def http_request(method, url_suffix, params_dict):
+        url = SERVER_URL + url_suffix
+
+        LOG('running %s request with url=%s\theaders=%s\nparams=%s' % (method, url, DEFAULT_HEADERS, json.dumps(params_dict)))
+
+        try:
+            res = requests.request(method,
+                                   url,
+                                   verify=USE_SSL,
+                                   params=params_dict,
+                                   headers=DEFAULT_HEADERS,
+                                  )
+            res.raise_for_status()
+
+            return res.json()
+
+        except Exception, e:
+            LOG(e)
+            raise(e)
+
+
+    def test_module():
+        try:
+            get_users_on_call_now_command()
+        except Exception, e:
+            raise Exception(e.message)
+
+        demisto.results("ok")
+
+    def extract_on_call_user_data(users):
+        """Extact data about user from a given schedule."""
+        outputs = []
```

Comment:
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

Reply:
Done